docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,196 @@
+
+== Configuration Options
+
+All the available configuration options can be found by either running the `help` command for individual commands:
+
+<@kc.all parameters="--help"/>
+
+Or by looking at all the configuration options at the <@link.guide title="All configuration"/> section.
+
+These options can be set using different formats:
+
+.Command-line argument
+<@kc.all parameters="--<category>-<sub-category>-<property>=<value>"/>
+
+.Environment variable
+```
+export KC_CATEGORY_SUB_CATEGORY_PROPERTY=<value>
+```
+
+.Properties at `conf/keycloak.properties`
+```
+<category>.<sub-category>.<property>=<value>
+```
+
+The format you should use is use-case specific but most of the time it should be enough to use the CLI. As an example, when running
+the server as a container you should be fine using both environment variables or command-line options.
+
+[NOTE]
+If you are unsure about the format of a specific property, consider looking at the <@link.guide title="All configuration"/> section.
+
+If the same configuration option is defined in different configuration sources, the effective value will be resolved by respecting the following order:
+
+. Command-line arguments
+. Environment variables
+. From `conf/keycloak.properties`
+
+== Configuring the server for optimal runtime
+
+Some configuration options are narrowed to a specific command or configuration stage. This clear separation is one of the key aspects
+of the server configuration and related to a series of optimizations that are done in order to deliver the best runtime when starting and running the server.
+
+By looking at the individual configuration options at this <@link.guide title="All configuration" text="section"/>, you will notice
+that some options require a `build`. In other words, they can only be set when running the `build` command.
+You can also check all the options that require a build by looking at the `help` message of the `build` command:
+
+<@kc.build parameters="--help"/>
+
+[NOTE]
+By default, the `help` message for the `build` command only shows the options that require a build. You can also use the `help-all` option to list all the available options, including those
+that do not require a build.
+
+The `build` command is responsible for producing an immutable and optimized server image (similar concept to building a container image). In addition to persisting
+any configuration option you have set, this command will also perform a series of optimizations to deliver the best runtime when starting and running the server. Basically,
+a lot of processing that would usually happen when starting/running the server is no longer necessary and the server can start and run faster.
+
+Some optimizations performed by the `build` command are:
+
+* Closed-world assumption about installed providers, hence no need to re-create the registry everytime the server starts
+* Configuration files (e.g.: clustering) are pre-parsed to reduce IO when starting the server
```

Comment:
Generally avoid latin: e.g.
This should either be "for example" or "in other words"